src/Control/Lens/Cons.hs

```diff
@@ -66,6 +67,10 @@ import qualified Data.Vector.Primitive as Prim
 import           Data.Vector.Unboxed (Unbox)
 import qualified Data.Vector.Unboxed as Unbox
 import           Data.Word
+#if __GLASGOW_HASKELL__ >= 708
+import           Data.Coerce
```

Comment:
To save yourself the need for CPP, import `Control.Lens.Internal.Coerce` instead. This exports a `coerce'` function which uses `coerce` on modern GHCs and `unsafeCoerce` on older ones.

Reply:
Changed to `coerce'`: https://github.com/ekmett/lens/pull/784/files#diff-484a1d6099769e258e439b127b5cfd6dR127